### PR TITLE
LPD-37898 | Wrong redirect link in the empty state of the Asset Appears On

### PR DESCRIPTION
**Pull request by rafaellajordao**
# Hi! :black_cat: 

### **Motivation:**
When we go to the overview page of an asset, when we access the card that appears on it, it is in the state of the empty card. The link that redirects to the documentation page is not correct.

### **Proposed solution:**
After talking to the product team, they agreed that the message with the redirect link to the documentation should be different for each type of asset. Thus changing the message and the link. The link should be to the specific paragraph dealing with Assets Appears On.

More details: [LPD-37898](https://liferay.atlassian.net/browse/LPD-37898)

### CODE REVIEW

**Comment by rafaellajordao** [comment]
All the tests run locally passed! :tada: 

```
Test Suites: 1 skipped, 614 passed, 614 of 615 total
Tests:       34 skipped, 2558 passed, 2592 total
Snapshots:   759 passed, 759 total
Time:        677.159 s
Ran all test suites.
Done in 677.95s.
```

**Comment by rafaellajordao** [comment]
ci:test:analytics-cloud

**Comment by rafaellajordao** [comment]
ci:test:sf

**Comment by rafaellajordao** [comment]
> **Jenkins Build:**[test-portal-acceptance-pullrequest(master)#11327](https://test-1-40.liferay.com/job/test-portal-acceptance-pullrequest(master)/11327/)
> **Jenkins Report:**[jenkins-report.html](https://test-1-40.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/11327/jenkins-report.html)
> **Jenkins Suite:**analytics-cloud
> **Pull Request:**[liferay-ac#1884](https://github.com/liferay-ac/liferay-portal/pull/1884)
> **Testray Routine:**[EE Pull Request](https://testray.liferay.com/#/project/630146/routines/630264)
> **Testray Build:**[[master] ci:test:analytics-cloud - rafaellajordao > liferay-ac - PR#1884 - 2024-10-11[05:15:35]](https://testray.liferay.com/#/project/630146/routines/630264/build/39275159)
> **Testray Build ID:**39275159
> **Testray Importer:**[publish-testray-report#24515](https://test-1-40.liferay.com/job/publish-testray-report/24515/)

the failures are not related to the changes made to the PR! :tada:

**Comment by rafaellajordao** [comment]
ci:test:relevant

**Comment by rafaellajordao** [comment]
ci:test:stable

File: modules/dxp/apps/osb/osb-faro/osb-faro-web/src/main/js/assets/components/AssetAppearsOnCard.tsx [hunk 1]
```diff
@@ -29,6 +29,15 @@ export enum Accessor {
 	PreviewsMetric = 'previewsMetric'
 }
 
+export enum EmptyStateLink {
+	Asset = URLConstants.AssetsCustomAssetsListDocumentation,
+	Blog = URLConstants.AssetsAppearsBlogsOnDocumentation,
+	Document = URLConstants.AssetsAppearsDocumentsAndMediaOnDocumentation,
+	Form = URLConstants.AssetsAppearsFormsOnDocumentation,
+	Journal = URLConstants.AssetsAppearsWebContentOnDocumentation,
+	WebContent = URLConstants.AssetsAppearsWebContentOnDocumentation
```
Comment (interaminense) [inline modules/dxp/apps/osb/osb-faro/osb-faro-web/src/main/js/assets/components/AssetAppearsOnCard.tsx:38]: Hi @rafaellajordao these two constants are the same, could you remove `journal`?

File: modules/dxp/apps/osb/osb-faro/osb-faro-web/src/main/js/assets/components/AssetAppearsOnCard.tsx [hunk 2]
```diff
@@ -38,6 +38,13 @@ export enum EmptyStateLink {
 	WebContent = URLConstants.AssetsAppearsWebContentOnDocumentation
 }
 
+export enum EmptyStateText {
+	Blog = Liferay.Language.get('learn-more-about-blogs'),
+	Document = Liferay.Language.get('learn-more-about-documents-and-media'),
+	Form = Liferay.Language.get('learn-more-about-forms'),
+	Journal = Liferay.Language.get('learn-more-about-web-content')
```
Comment (interaminense) [inline modules/dxp/apps/osb/osb-faro/osb-faro-web/src/main/js/assets/components/AssetAppearsOnCard.tsx:45]: Can you rename it to WebContent?

File: modules/dxp/apps/osb/osb-faro/osb-faro-web/src/main/js/assets/web-content/pages/Overview.tsx [hunk 3]
```diff
@@ -56,6 +58,8 @@ const Overview = () => (
 				<AssetAppearsOnCard
 					accessors={[Accessor.ViewsMetric]}
 					assetType={AssetTypes.Journal}
+					emptyStateLink={EmptyStateLink.Journal}
+					emptyStateText={EmptyStateText.Journal}
```
Comment (interaminense) [inline modules/dxp/apps/osb/osb-faro/osb-faro-web/src/main/js/assets/web-content/pages/Overview.tsx:62]: Can you use .WebContent instead?

**Review by interaminense — CHANGES_REQUESTED** [review]
Hi @rafaellajordao, I just left few changes, thanks!

File: modules/dxp/apps/osb/osb-faro/osb-faro-web/src/main/js/assets/components/AssetAppearsOnCard.tsx (hunk 1 above)
Comment (rafaellajordao) [reply modules/dxp/apps/osb/osb-faro/osb-faro-web/src/main/js/assets/components/AssetAppearsOnCard.tsx]: removed!

**Comment by rafaellajordao** [comment]
ci:test:sf

**Comment by rafaellajordao** [comment]
ci:test:relevant

**Comment by rafaellajordao** [comment]
ci:test:stable

**Comment by rafaellajordao** [comment]
All the tests run locally passed! 🎉
```
Test Suites: 1 skipped, 614 passed, 614 of 615 total
Tests:       34 skipped, 2558 passed, 2592 total
Snapshots:   759 passed, 759 total
Time:        622.003 s
Ran all test suites.
Done in 622.80s.
```

**Comment by rafaellajordao** [comment]
ci:test:stable

**Comment by rafaellajordao** [comment]
ci:test:relevant

**Comment by rafaellajordao** [comment]
> the failures are not related to the changes made to the PR! 🎉

the failures are not related to the changes made to the PR! 🎉

**Comment by rafaellajordao** [comment]
> ### ❌ ci:test:analytics-cloud - 243 out of 253 jobs passed in 3 hours 9 minutes
> Click here for more details.

the failures are not related to the changes made to the PR! 🎉

**Comment by rafaellajordao** [comment]
> ### ❌ ci:test:stable - 26 out of 27 jobs passed
> ### ❌ ci:test:relevant - 33 out of 36 jobs passed in 1 hour 43 minutes
> Click here for more details.

hey @interaminense , do you know why these tests are failing?

**Comment by interaminense** [comment]
It is not related to your changes, it is failing in other PR's

**Comment by interaminense** [comment]
ci:test:stable

**Comment by interaminense** [comment]
If the stable passes, we can move it to Brian.

**Review by interaminense — APPROVED** [review]
LGTM! Thanks @rafaellajordao :tada:

**Comment by rafaellajordao** [comment]
ci:forward

**Comment by interaminense** [comment]
ci:test:stable

**Comment by interaminense** [comment]
ci:test:stable

**Comment by interaminense** [comment]
ci:forward:force

**Pull request closed without merging**